doc/modules/gaussian_process.rst

```diff
@@ -8,45 +8,514 @@ Gaussian Processes
+fitted for each class, which is trained to separate this class from the rest.
+In "one_vs_one", one binary Gaussian process classifier is fitted for each pair
+of classes, which is trained to separate these two classes. The predictions of
+these binary predictors are combined into multi-class predictions. See the
+section on :ref:`multi-class classification <multiclass>` for more details.
+
+In the case of Gaussian process classification, "one_vs_one" might be
+computationally  cheaper since it has to solve many problems involving only a
+subset of the whole training set rather than fewer problems on the whole
+dataset. Since Gaussian process classification scales cubically with the size
+of the dataset, this might be considerably faster. However, note that
+"one_vs_one" does not support predicting probability estimates but only plain
+predictions. Moreover, note that :class:`GaussianProcessClassifier` does not
+(yet) implement a true multi-class Laplace approximation, but is based on
+several binary classification tasks.
+
+GPC examples
+============
+
+Probabilistic predictions with GPC
+----------------------------------
+
+This example illustrates the predicted probability of GPC for an RBF kernel
+with different choices of the hyperparameters. The first figure shows the
+predicted probability of GPC with arbitrarily chosen hyperparameters and with
+the hyperparameters corresponding to the maximum log-marginal-likelihood (LML).
+
+While the hyperparameters chosen by optimizing LML have a considerable larger
+LML, they perform slightly worse according to the log-loss on test data. The
+figure shows that this is because they exhibit a steep change of the class
+probabilities at the class boundaries (which is good) but have predicted
+probabilities close to 0.5 far away from the class boundaries (which is bad)
+This undiesirable effect is caused by the Laplace approximation used
+internally by GPC.
+
+The second figure shows the log-marginal-likelihood for different choices of
+the kernel's hyperparameters, highlighting the two choices of the
+hyperparameters used in the first figure by black dots.
+
+.. figure:: ../auto_examples/gaussian_process/images/plot_gpc_000.png
+   :target: ../auto_examples/gaussian_process/plot_gpc.html
+   :align: center
+
+.. figure:: ../auto_examples/gaussian_process/images/plot_gpc_001.png
+   :target: ../auto_examples/gaussian_process/plot_gpc.html
+   :align: center
+
+
+Illustration of GPC on the XOR dataset
+--------------------------------------
+
+.. currentmodule:: sklearn.gaussian_process.kernels
+
+This example illustrates GPC on XOR data. Compared are a stationary, isotropic
+kernel (`RBF`) and a non-stationary kernel (`DotProduct`). On this particular
+dataset, the `DotProduct` kernel obtains considerably better results because the
+class-boundaries are linear and coincide with the coordinate axes. In general,
+stationary kernels often obtain better results.
```

Comment:
In general -> In practice, however, stationary kernels such as `RBF`often obtain better results.